test/test_torch.py

```diff
@@ -10498,6 +10498,11 @@ def check(op, a, args, key):
         check(torch.median, [[nan, nan], [1, 2]], [1], [[nan, 1]])
         check(torch.nanmedian, [[nan, nan], [1, 2]], [1], [[nan, 1.]])
 
+        # Discontiguous and strided tensors
```

Comment:
Try a 1D 2D and 3D test case. 1D can be every other element, 2D and 3D can be transposed. For 2D and 3D also try selecting the median along a dimension in addition to flattening (like you're doing here in the 2D case. You can compare against NumPy, too, if you don't want to work out the fixtures.

Follow-up question, we've had two of these issues now. One for median and another for kth value (https://github.com/pytorch/pytorch/pull/45802). These bugs were in PyTorch for awhile, too (since 1.6 at least). Should we create a test that iterates over a variety of these ops? kth value, median, nanmedian, min, max, argmin, argmax, sum...?

Reply:
I believe the issue was due to this subroutine called by both median and kthvalue which computes the data offsets assuming contiguous input. 
https://github.com/pytorch/pytorch/blob/daf2a6a29da5109157bb44fc5e4dbc4fb5c56c15/aten/src/ATen/native/cuda/SortingCommon.cuh#L147

Reply:
Is that subroutine called by anything else?

Reply:
no

Reply:
Cool; let's add a WARNING comment there that the input must be contiguous.

Reply:
min, max, argmin, armax, sum are tested for discontiguous inputs.

Reply:
@heitorschueroff isn't this subroutine also called for nanmedian?

Reply:
@ngimel, yes and that's taken care of in this PR since median and nanmedian call median_with_indices_impl